flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/CatalogManager.java

```diff
@@ -477,6 +478,41 @@ public void createTemporaryTable(
 		});
 	}
 
+	/**
+	 * Qualifies the given {@link UnresolvedIdentifier} with current catalog & database and
+	 * removes a temporary table registered with this path if it exists.
+	 *
+	 * @param identifier potentially unresolved identifier
+	 * @return true if a table with a given identifier existed and was removed, false otherwise
+	 */
+	public boolean dropTemporaryTable(UnresolvedIdentifier identifier) {
+		return doDropTemporaryTable(identifier, (table) -> table instanceof CatalogTable);
+	}
+
+	/**
+	 * Qualifies the given {@link UnresolvedIdentifier} with current catalog & database and
+	 * removes a temporary view registered with this path if it exists.
+	 *
+	 * @param identifier potentially unresolved identifier
+	 * @return true if a view with a given identifier existed and was removed, false otherwise
+	 */
+	public boolean dropTemporaryView(UnresolvedIdentifier identifier) {
+		return doDropTemporaryTable(identifier, (table) -> table instanceof CatalogView);
+	}
+
+	private boolean doDropTemporaryTable(
```

Comment:
the API naming of "doXxx()" feels a bit weird. rename to `dropTemporaryTableInternal()`?